llmfoundry/train/train.py

```diff
@@ -0,0 +1,580 @@
+        progress_bar=train_cfg.progress_bar,
+        log_to_console=train_cfg.log_to_console,
+        console_log_interval=train_cfg.console_log_interval,
+        loggers=loggers,
+        callbacks=callbacks,
+        precision=train_cfg.precision,
+        algorithms=algorithms,
+        device_train_microbatch_size=train_cfg.device_train_microbatch_size,
+        parallelism_config={'fsdp': fsdp_config},
+        save_folder=train_cfg.save_folder,
+        save_filename=save_filename,
+        save_latest_filename=save_latest_filename,
+        save_interval=train_cfg.save_interval,
+        save_num_checkpoints_to_keep=train_cfg.save_num_checkpoints_to_keep,
+        save_overwrite=train_cfg.save_overwrite,
+        save_weights_only=train_cfg.save_weights_only,
+        load_path=train_cfg.load_path,
+        load_weights_only=train_cfg.load_weights_only,
+        load_strict_model_weights=train_cfg.load_strict_model_weights,
+        load_ignore_keys=train_cfg.load_ignore_keys,
+        save_ignore_keys=train_cfg.save_ignore_keys,
+        autoresume=train_cfg.autoresume,
+        python_log_level=train_cfg.python_log_level,
+        dist_timeout=train_cfg.dist_timeout,
+        profiler=profiler,
+        compile_config=compile_config,
+    )
+
+    # Optionally just save an HF checkpoint
+    if train_cfg.only_hf_checkpoint:
+        hf_checkpointer_callbacks = [
+            c for c in callbacks if isinstance(c, HuggingFaceCheckpointer)
+        ]
+        if len(hf_checkpointer_callbacks) == 0:
+            raise ValueError(
+                'No HuggingFaceCheckpointer callback found, but only_hf_checkpoint was set to True. Please add a HuggingFaceCheckpointer.',
+            )
+        if len(hf_checkpointer_callbacks) > 1:
+            raise ValueError(
+                'Multiple HuggingFaceCheckpointer callbacks found, but only_hf_checkpoint was set to True. Please remove all but one HuggingFaceCheckpointer.',
+            )
+
+        hf_checkpointer_callback = hf_checkpointer_callbacks[0]
+        hf_checkpointer_callback._save_checkpoint(trainer.state, trainer.logger)
+        return trainer
+
+    if train_cfg.log_config:
+        log.info('Logging config')
+        log_config(logged_cfg)
+    log_dataset_uri(logged_cfg)
+    torch.cuda.empty_cache()
+    gc.collect()
+
+    # Eval first if requested
+    if train_cfg.eval_first and trainer.state.timestamp.batch.value == 0:
+        trainer.eval()
+
+    log.info('Starting training...')
+    trainer.fit()
+
+    log.info('Done.')
+    return trainer
+
+
+if __name__ == '__main__':
+    yaml_path, args_list = sys.argv[1], sys.argv[2:]
+
+    # Disable resolving environment variables through omegaconf.
+    om.clear_resolver('oc.env')
+
+    # Load yaml and cli arguments.
+    with open(yaml_path) as f:
+        yaml_cfg = om.load(f)
+    cli_cfg = om.from_cli(args_list)
+    cfg = om.merge(yaml_cfg, cli_cfg)
+    assert isinstance(cfg, DictConfig)
+    train(cfg)
```

Comment:
remove? This doesn't need to be a script right?

Reply:
Doesn't need to be in the script